wpilibc/shared/src/Commands/CommandGroup.cpp

```diff
@@ -158,21 +153,20 @@ void CommandGroup::AddParallel(Command* command, double timeout) {
     return;
   }
 
+  m_commands.emplace_back(command, CommandGroupEntry::kSequence_BranchChild,
+                          timeout);
+
   command->SetParent(this);
 
-  m_commands.push_back(CommandGroupEntry(
-      command, CommandGroupEntry::kSequence_BranchChild, timeout));
   // Iterate through command->GetRequirements() and call Requires() on each
   // required subsystem
-  Command::SubsystemSet requirements = command->GetRequirements();
-  auto iter = requirements.begin();
-  for (; iter != requirements.end(); iter++) Requires(*iter);
+  for (auto requirement : command->GetRequirements()) Requires(requirement);
 }
 
 void CommandGroup::_Initialize() { m_currentCommandIndex = -1; }
 
 void CommandGroup::_Execute() {
-  CommandGroupEntry entry;
+  CommandGroupEntry* entry;
```

Comment:
It's a non-owning pointer to elements of std::list<CommandGroupEntry*> m_children, which is itself a list of non-owning pointers. This used to be part of a patch that fixed Command's memory leaks with std::unique_ptr (Command-based robot leaks memory every time a Command is created).

I separate this from it so it could be merged separately. The latter part wasn't merged because actually enforcing memory ownership constraints will break user code that frees the memory itself. It's still on my personal mirror.